server/src/index.js

```diff
@@ -83,38 +83,41 @@ function normalizeEmoji(emoji) {
   return res
 }
 
-app.get('/api/reports-by-tags/:tag', authorize, async (req, res) => {
-  let reports = await loadReportsByTag(req.params.tag)
+async function loadReportData() {
+  let reports = await loadReports()
   reports = await Promise.all(reports.map(async (r) => ({
     ...r,
-    replies: (
-      await cache.get(
-        replies, r.response_to || r.ts, r.channel
-      )).messages,
+    replies: [...await loadReplies(r.response_to || r.ts)],
```

Comment:
This point is my only concern about this solution. We are hitting the database nearly 500 times, which is not nice. 

Do you think we can optimize this? 

A really easy change would be for `loadReplies` to load everything (sorted by `ts` as it is now) and go:

```
  const repliesByThread = _.groupBy(await loadReplies(), (r) => r.response_to || r.ts)
  let reports = (await loadReports()).map((r) => ({
    ...r,
    replies: repliesByThread[r.response_to || r.ts]
  }))
```